### PR TITLE
Make PageObjectElement.getText() async

### PR DESCRIPTION
**Pull request by dskloetd**
# Motivation

We want to share page objects between unit tests and e2e tests.
In e2e tests some things have to be async because they involve making a request to the browser driver.
So for compatibility they have to be async in the platform agnostic interface.
One such things is getting the text content of an element.

# Changes

1. Make PageObjectElement.getText() async.
2. Make everything derived from it async as well.

# Tests

npm run test
npm run check

### CODE REVIEW

File: frontend/src/tests/page-objects/AmountDisplay.page-object.ts [hunk 1]
```diff
@@ -15,8 +15,9 @@ export class AmountDisplayPo {
     return el && new AmountDisplayPo(el);
   }
 
-  getAmount(): string | null {
-    return assertNonNullish(this.root.querySelector(`[data-tid="token-value"]`))
-      .textContent;
+  getAmount(): Promise<string> {
```
Comment (lmuntaner) [inline frontend/src/tests/page-objects/AmountDisplay.page-object.ts:18]: `getText` doesn't return `null` now?

**Review by lmuntaner — APPROVED** [review]
Thanks! Approving, but I added a small question

File: frontend/src/tests/page-objects/AmountDisplay.page-object.ts (hunk 1 above)
Comment (dskloetd) [reply frontend/src/tests/page-objects/AmountDisplay.page-object.ts:18]: textContent/getText() was never null.
I think this method could return null before the assertNonNullish was added and the annotation wasn't updated.

Comment (lmuntaner) [reply frontend/src/tests/page-objects/AmountDisplay.page-object.ts:18]: Ok, thanks, that makes sense

**Pull request merged**